### ISSUE TITLE
Overloaded method marked @final does not generate error when overridden

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6860]
The following should result in a type error but does not.

```python
from typing import final, overload

class Base:
    @overload
    def method(self, x: int) -> int:
        ...

    @overload
    def method(self, x: str) -> str:
        ...

    @final
    def method(self, x: int | str) -> int | str:
        ...

class Derived(Base):
    @overload
    def method(self, x: int) -> int:
        ...

    @overload
    def method(self, x: str) -> str:
        ...

    def method(self, x: int | str) -> int | str:  # Type error
        ...
```

This should also work with stubs where the first overload is marked @final.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
